visidata/aggregators.py

```diff
@@ -142,11 +143,39 @@ def __init__(self, pct, helpstr=''):
     def aggregate(self, col, rows):
         return _percentile(sorted(col.getValues(rows)), self.pct/100, key=float)
 
+class RankAggregator(Aggregator):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def aggregate(self, col, rows):
+        if not col.sheet.keyCols:
+            vd.error('ranking requires one or more key columns')
```

Comment:
Is this actually true?  I could see row number being used if there are no key columns.  If we remove this check, does that just work?

Reply:
Using row number for ranking does the same thing as `addcol-incr`. Would it work to make `rank()`, when there are no key columns, act as if all columns are key columns?

If so, should it use only visible columns, or all columns including hidden ones?

Reply:
My questions above are obsolete, see https://github.com/saulpw/visidata/pull/2417#discussion_r1681972604 for newer context.